### ISSUE TITLE
about training

### ISSUE THREAD

**Issue by hktxt** [adambielski/siamese-triplet#16]
Hi, I have some questions about siamese network.
Siamese network has two branch of shared-weight net, right? It has two input, so I can feed a pair of data(one Pos and one Neg) to the net. How to treat the triplet case, one Anchor, one P and one N, how to feed to the siamese net? three data VS two input channel...
Another one is, Siamese has two weight-shared nets, does this means that the two net have the same weight, if two nets have the same architecture ? If same architecture, same weight, It can be done using one net to compare the difference...
The last one is about the training, Some papers say that the contrastive loss is not easy to train, so pre-trained using softmax, and then training using contrastive loss. Their siamese net pre-trained using softmax. But what confused me is that how the classification problem trained using siamese net with softmax. May be a single net trained using softmax, and then combined together as siamese? Only in this way I can understand.
any responses are welcomed, thanks ~~~
